### PR TITLE
added users password cryptography

### PR DESCRIPTION
**Pull request by GuchiTheHasky**
- added user password cryptography;
- added abiity to change user's password for administrator;
- fix error in TestConfigurationToCountAllQueries;

### CODE REVIEW

**Review by radeon323 — APPROVED** [review]

File: goods-partner-backend/src/main/java/com/goodspartner/exception/CryptographyException.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package com.goodspartner.exception;
+
+public class CryptographyException extends RuntimeException {
```
Comment (belosh59) [inline goods-partner-backend/src/main/java/com/goodspartner/exception/CryptographyException.java:3]: This class is not used anywhere

File: goods-partner-backend/src/main/java/com/goodspartner/mapper/UserMapper.java [hunk 2]
```diff
@@ -11,6 +11,7 @@ public interface UserMapper {
 
     UserDto toUserDto(User user);
 
+    @Mapping(source = "password", target = "password")
```
Comment (belosh59) [inline goods-partner-backend/src/main/java/com/goodspartner/mapper/UserMapper.java:14]: If there equal attributes mapstruct automatically match them, no need to separately define them

Comment (belosh59) [reply goods-partner-backend/src/main/java/com/goodspartner/mapper/UserMapper.java:14]: you could od clean compile and check the generated code

Comment (belosh59) [reply goods-partner-backend/src/main/java/com/goodspartner/mapper/UserMapper.java:14]: Screenshot 2023-11-24 at 20 33 11

File: goods-partner-backend/src/main/java/com/goodspartner/service/impl/DefaultCryptographer.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    private final PasswordEncoder passwordEncoder;
+
+    @Value("${salt}")
+    private String salt;
+    public String encode(CharSequence rawPassword) {
+        String passwordWithSalt = rawPassword + salt;
+        return passwordEncoder.encode(passwordWithSalt);
+    }
+
+    public boolean matches(CharSequence rawPassword, String encodedPassword) {
```
Comment (belosh59) [inline goods-partner-backend/src/main/java/com/goodspartner/service/impl/DefaultCryptographer.java:22]: this method is not used anywhere

File: goods-partner-backend/src/main/java/com/goodspartner/service/impl/DefaultCryptographer.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class DefaultCryptographer implements Cryptographer {
+
+    private final PasswordEncoder passwordEncoder;
```
Comment (belosh59) [inline goods-partner-backend/src/main/java/com/goodspartner/service/impl/DefaultCryptographer.java:13]: What PasswordEncoder implementation is used here?
Coz there a lot of different

We are using this one:
com.goodspartner.configuration.security.AuthenticationManagerConfig#passwordEncoder
BCryptPasswordEncoder

File: goods-partner-backend/src/main/java/com/goodspartner/service/impl/DefaultCryptographer.java (hunk 3 above)
Comment (GuchiTheHasky) [reply goods-partner-backend/src/main/java/com/goodspartner/service/impl/DefaultCryptographer.java:22]: Fixed.

File: goods-partner-backend/src/main/java/com/goodspartner/mapper/UserMapper.java (hunk 2 above)
Comment (GuchiTheHasky) [reply goods-partner-backend/src/main/java/com/goodspartner/mapper/UserMapper.java:14]: Fixed.

File: goods-partner-backend/src/main/java/com/goodspartner/exception/CryptographyException.java (hunk 1 above)
Comment (GuchiTheHasky) [reply goods-partner-backend/src/main/java/com/goodspartner/exception/CryptographyException.java:3]: Fixed.

File: goods-partner-backend/src/main/java/com/goodspartner/service/impl/DefaultCryptographer.java (hunk 4 above)
Comment (GuchiTheHasky) [reply goods-partner-backend/src/main/java/com/goodspartner/service/impl/DefaultCryptographer.java:13]: Sorry, already fixed.

**Pull request merged**